lib/ocpp/v201/charge_point.cpp

```diff
@@ -3194,6 +3209,178 @@ void ChargePoint::handle_customer_information_req(Call<CustomerInformationReques
+            EVLOG_warning << "Could not persist in device model new monitoring level: " << msg.severity;
+            response.status = GenericStatusEnum::Rejected;
+        } else {
+            response.status = GenericStatusEnum::Accepted;
+        }
+    }
+
+    ocpp::CallResult<SetMonitoringLevelResponse> call_result(response, call.uniqueId);
+    this->send<SetMonitoringLevelResponse>(call_result);
+}
+
+void ChargePoint::handle_set_variable_monitoring_req(Call<SetVariableMonitoringRequest> call) {
+    SetVariableMonitoringResponse response;
+    const auto& msg = call.msg;
+
+    try {
+        response.setMonitoringResult = this->device_model->set_monitors(msg.setMonitoringData);
+    } catch (const ocpp::common::QueryExecutionException& e) {
+        EVLOG_error << "Set monitors failed:" << e.what();
+    }
+
+    ocpp::CallResult<SetVariableMonitoringResponse> call_result(response, call.uniqueId);
+    this->send<SetVariableMonitoringResponse>(call_result);
+}
+
+void ChargePoint::notify_monitoring_report_req(const int request_id,
+                                               const std::vector<MonitoringData>& montoring_data) {
+    static constexpr int32_t MAXIMUM_VARIABLE_SEND = 10;
+
+    if (montoring_data.size() <= MAXIMUM_VARIABLE_SEND) {
+        NotifyMonitoringReportRequest req;
+        req.requestId = request_id;
+        req.seqNo = 0;
+        req.generatedAt = ocpp::DateTime();
+        req.monitor.emplace(montoring_data);
+        req.tbc = false;
+
+        ocpp::Call<NotifyMonitoringReportRequest> call(req, this->message_queue->createMessageId());
+        this->send<NotifyMonitoringReportRequest>(call);
+    } else {
+        // Split for larger message sizes
+        int32_t sequence_num = 0;
+        auto generated_at = ocpp::DateTime();
+
+        for (int32_t i = 0; i < montoring_data.size(); i += MAXIMUM_VARIABLE_SEND) {
+            // If our next index is >= than the last index then we're finished
+            bool last_part = ((i + MAXIMUM_VARIABLE_SEND) >= montoring_data.size());
+
+            NotifyMonitoringReportRequest req;
+            req.requestId = request_id;
+            req.seqNo = sequence_num;
+            req.generatedAt = generated_at;
+            req.tbc = (!last_part);
+
+            // Construct sub-message part
+            std::vector<MonitoringData> sub_data;
+
+            for (int32_t j = i; j < MAXIMUM_VARIABLE_SEND && j < montoring_data.size(); ++j) {
+                sub_data.push_back(std::move(montoring_data[i + j]));
+            }
+
+            req.monitor = sub_data;
+
+            ocpp::Call<NotifyMonitoringReportRequest> call(req, this->message_queue->createMessageId());
+            this->send<NotifyMonitoringReportRequest>(call);
+
+            sequence_num++;
+        }
+    }
+}
+
+void ChargePoint::handle_get_variable_monitoring_req(Call<GetMonitoringReportRequest> call) {
+    GetMonitoringReportResponse response;
+    const auto& msg = call.msg;
+
```

Comment:
You can implement `N02.FR.07` similar to https://github.com/EVerest/libocpp/blob/main/lib/ocpp/v201/charge_point.cpp#L2295